maven-plugin-annotations/src/main/java/org/apache/maven/plugins/annotations/Parameter.java

```diff
@@ -72,6 +72,13 @@
      */
     String defaultValue() default "";
 
+    /**
+     * Defines the implementation in the case the parameter type is an interface.
+     *
+     * @return the implementation class name
+     */
+    Class<?> implementation() default Object.class;
```

Comment:
Question: what if the class will only be available at runtime by the user, how can it be provided here? The previous code used strings, no?

Reply:
When user uses annotation in their code, they have access to all classes. 

In plugin-tools we operate on Type from asm, and finally on class name as string, so we needn't access to class.

I hope that define property as class will be less error-prone from user perspective.

The previous implementation was only for javadoc tags ... so must be string.

Reply:
So the rationale is that if you know that you need a specific implementation you need to have it in `compile` scope?

Reply:
The same is in https://github.com/apache/maven-plugin-tools/blob/master/maven-plugin-annotations/src/main/java/org/apache/maven/plugins/annotations/Component.java

Reply:
I don't agree here. The `role` is the interface, no? The hint is the named impl as a string. I don't think that it compares. Am I wrong?

Reply:
Ok, I think it is never mind (probably) if implementation will be string or class for `parameter` annotation is used in plugin itself not in plugin configuration.
Finally class must be on plugin classpath.
 
We also have `implementation` attribute for configuration ... https://maven.apache.org/guides/mini/guide-configuring-plugins.html#mapping-complex-objects

Now I don't know which one wins implementation from parameter annotation / javadoc tags ... or from configuration ... 

😄

Reply:
@hboutemy Do you have an opinion on this?

Reply:
Correct, tricky one... @cstamas WDYT?

Reply:
currently when user use javadoc tags can deliver such information ... of course as string without verification on code level ...

Reply:
OK, I'd still to `Class`, but like to ultimately here @cstamas on this.